### PR TITLE
fix(e2e): cleanup after each test

### PR DESCRIPTION
**Pull request by Pothulapati**

### CODE REVIEW

File: e2e/dragonfly_controller_test.go [hunk 1]
```diff
@@ -472,6 +472,18 @@ var _ = Describe("Dragonfly Lifecycle tests", Ordered, func() {
...
+
+		It("Cleanup", func() {
+			var df dragonflydbiov1alpha1.Dragonfly
+			err := k8sClient.Get(ctx, types.NamespacedName{
+				Name:      name,
+				Namespace: namespace,
+			}, &df)
+			Expect(err).To(BeNil())
+
+			err = k8sClient.Delete(ctx, &df)
```
Comment (Abhra303) [inline e2e/dragonfly_controller_test.go:483]: Should we also check if the related resources are also deleted (i.e. deployments, service etc)?

File: e2e/dragonfly_pod_lifecycle_controller_test.go [hunk 2]
```diff
@@ -22,6 +22,7 @@ import (
 	"time"
 
 	dfv1alpha1 "github.com/dragonflydb/dragonfly-operator/api/v1alpha1"
+	dragonflydbiov1alpha1 "github.com/dragonflydb/dragonfly-operator/api/v1alpha1"
```
Comment (Abhra303) [inline e2e/dragonfly_pod_lifecycle_controller_test.go]: Why do we need the same import with different import name (`dfv1alpha1` and `dragonflydbiov1alpha1`?

**Review by Abhra303 — APPROVED** [review]
LGTM with some minor comments

File: e2e/dragonfly_controller_test.go (hunk 1 above)
Comment (Pothulapati) [reply e2e/dragonfly_controller_test.go:483]: Deletion is not the purview of the Operator. It is done automatically by the garbage collector in k8s. This is through the Owner reference that we pass to resources.

Comment (Abhra303) [reply e2e/dragonfly_controller_test.go:483]: Agreed

**Pull request merged**